rust/parquet/src/column/writer.rs

```diff
@@ -387,15 +538,28 @@ impl<T: DataType> ColumnWriterImpl<T> {
             ));
         }
 
-        // TODO: update page statistics
+        if calculate_page_stats {
+            for val in &values[0..values_to_write] {
+                if self.min_page_value.is_none()
```

Comment:
maybe we should extract this into a util function `update_page_min_max` and similarly for column min/max value.